### PR TITLE
Fixed typos in several doc files

### PR DESCRIPTION
**Pull request by Lunalicious**
Fixed typos in:
- c_sharp_differences.rst
- debugger_panel.rst
- gdextension_cpp_example.rst
- gui_using_fonts.rst
- openxr_hand_tracking.rst
- overview_of_debugging_tools.rst
- setting_up_xr.rst
- shading_language.rst
- the_profiler.rst
- your_second_3d_shader.rst

### CODE REVIEW

File: tutorials/scripting/debug/the_profiler.rst [hunk 1]
```diff
@@ -72,7 +72,7 @@ Physics, and Audio, before reaching Script functions, where your code appears.
 
 If you click on the graph, you change which frame's information appears on the
 left. In the top right, there is also a frame counter where you can manually
-adjust the frame you are looking at more granularly.
+adjust the frame you are looking at more granular.
```
Comment (AThousandShips) [inline tutorials/scripting/debug/the_profiler.rst:75]: This is a [real word](https://en.wiktionary.org/wiki/granularly#English), not necessarily a typo

File: tutorials/shaders/shader_reference/shading_language.rst [hunk 2]
```diff
@@ -1310,7 +1310,7 @@ is used, it can be scalar or vector.
 | vec4 **textureGrad** (samplerCubeArray s, vec3 p, vec3 dPdx,                |                                                                     |
 | vec3 dPdy)                                                                  |                                                                     |
 +-----------------------------------------------------------------------------+---------------------------------------------------------------------+
-| gvec4_type **textureProjGrad** (gsampler2D s, vec3 p, vec2 dPdx, vec2 dPdy) | Performs a texture read with projection/LOD and with explcit        |
+| gvec4_type **textureProjGrad** (gsampler2D s, vec3 p, vec2 dPdx, vec2 dPdy) | Performs a texture read with projection/LOD and with explicit        |
```
Comment (AThousandShips) [inline tutorials/shaders/shader_reference/shading_language.rst:1313]:
```suggestion
| gvec4_type **textureProjGrad** (gsampler2D s, vec3 p, vec2 dPdx, vec2 dPdy) | Performs a texture read with projection/LOD and with explicit       |
```

File: tutorials/scripting/debug/the_profiler.rst (hunk 1 above)
Comment (Lunalicious) [reply tutorials/scripting/debug/the_profiler.rst]: Oh yes, sorry! Thanks for pointing that out

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Merged. Quite a few typos fixed, great! Thanks and congrats on your first merged contribution.

**Comment by mhilbrunner** [comment]
Backported to 3.6 in https://github.com/godotengine/godot-docs/pull/9937.

**Pull request merged**